src/Symfony/Component/HttpFoundation/StreamedJsonResponse.php

```diff
@@ -0,0 +1,149 @@
+
+use function array_keys;
+use function explode;
+use function flush;
+
+use Generator;
+
+use function json_encode;
+
+use const JSON_THROW_ON_ERROR;
+
+/**
+ * StreamedJsonResponse represents a streamed HTTP response for JSON.
+ *
+ * A StreamedJsonResponse uses a structure and generics to create an
+ * efficient resource saving JSON response.
+ *
+ * It uses flush after a specified flush size to directly stream the data.
+ *
+ * @see flush()
+ *
+ * @author Alexander Schranz <alexander@sulu.io>
+ *
+ * Example usage:
+ *
+ * $response = new StreamedJsonResponse(
+ *     // json structure with replace identifiers
+ *     [
+ *         '_embedded' => [
+ *             'articles' => '__articles__',
+ *         ],
+ *     ],
+ *     // array of generators with replace identifier used as key
+ *     [
+ *         '__articles__' => function (): \Generator { // any method or function returning a Generator
+ *              yield ['title' => 'Article 1'];
+ *              yield ['title' => 'Article 2'];
+ *              yield ['title' => 'Article 3'];
+ *         },
+ *     ]
+ * );
+ */
+class StreamedJsonResponse extends StreamedResponse
+{
+    public const DEFAULT_ENCODING_OPTIONS = JsonResponse::DEFAULT_ENCODING_OPTIONS;
+
+    protected int $encodingOptions = self::DEFAULT_ENCODING_OPTIONS;
+
+    /**
+     * @param mixed[] $structure
+     * @param array<string, Generator<int|string, mixed> $generics
+     */
+    public function __construct(
+        private readonly array $structure,
+        private readonly array $generics,
+        int $status = 200,
+        array $headers = [],
+        private int $flushSize = 500,
+    ) {
+        parent::__construct(function () {
+            $keys = array_keys($this->generics);
+
+            $jsonEncodingOptions = JSON_THROW_ON_ERROR | $this->getEncodingOptions();
+            $structureText = json_encode($this->structure, $jsonEncodingOptions);
+
+            foreach ($keys as $key) {
+                [$start, $end] = explode('"'.$key.'"', $structureText, 2);
+
+                // send first and between parts of the structure
+                echo $start;
+
+                $count = 0;
+                echo '[';
+                foreach ($this->generics[$key] as $array) {
+                    if (0 !== $count) {
+                        // if not first element of the generic a separator is required between the elements
+                        echo ',';
+                    }
+
+                    echo json_encode($array, $jsonEncodingOptions);
+                    ++$count;
+
+                    if (0 === $count % $this->flushSize) {
+                        flush();
+                    }
+                }
+                echo ']';
+
+                $structureText = $end;
+            }
+
+            echo $structureText; // send the after part of the structure as last
+        }, $status, $headers);
+
+        $this->headers->set('Content-Type', 'application/json');
```

Comment:
Should be called before `parent::__construct()` to allow `$headers` to override it if it defines the content type?

Reply:
Good catch, sadly the headers are not initialized then and can not be set before the parent::__construct but checked if content-type is set and only set it then. Added also a test case for it.

Reply:
resolving this as it was fixed :)